app/controllers/books_controller.rb

```diff
@@ -20,6 +20,19 @@ def create
     end
   end
 
+  def edit
+    @book = Book.find(params[:id])
+  end
+
+  def update
+    @book = Book.find(params[:id])
+    if @book.update(book_params)
```

Comment:
We can see here that we repeatedly find a book in `show`, `edit`, and `update` to make our code more DRY we can use `before_action`.

Example: https://dannemanne.com/posts/ruby_on_rails_explained_before_action